.github/workflows/add-netlify-links.yml

```diff
@@ -1,55 +1,75 @@
 name: Add Netlify Links To Changed Pages
 on:
-    workflow_call:
+  workflow_call:
+  pull_request_target:
 jobs:
-    get-pr-changes:
-        name: Get Changed Files & Update PR Description
-        runs-on: ubuntu-latest
-        permissions:
-            issues: write
-            contents: write
-            pull-requests: write
-            repository-projects: write
-        steps:
-          - uses: actions/checkout@v4
-          - name: Get Changed Files
-            id: changed-files
-            uses: tj-actions/changed-files@v44
-            with:
-              separator: ","
-              files: source/**
-          - name: Build Netlify Links for Changed Pages
-            id: build_page_links
-            run: |
-              new_links=""
-              base_link='https://deploy-preview-${{ github.event.number }}--mongodb-docs-csharp.netlify.app'
-              changed_files=${{ steps.changed-files.outputs.all_changed_files }}
-              files=$(echo $changed_files | tr "," "\n")
-              for file in $files; do
-                echo "processing ${file}"
-                if (! grep -s "includes/"  <<< $file) && 
-                    (! grep -s "images/"  <<< $file) && 
-                    (! grep -s "examples/"  <<< $file); then
-                  file="${file#source}"
-                  file="${file%.txt}"
-                  filenoslash="${file:1}"
-                  echo "${base_link}${file}"
-                  new_links+="<li><a href=${base_link}${file}>${filenoslash}</a></li>"
-                else
-                  echo "(file skipped)"
-                fi
-              done
-              if [ "$new_links" == "" ]; then
-                new_links="No pages to preview"
-              fi
-              echo "Final new_links string: "
-              echo "${new_links}"
-              echo "staging_links=${new_links}" >> "$GITHUB_OUTPUT"
-          - name: Update the PR Description
-            uses: MongoCaleb/pr-description-action@master
-            with:
-                regex: "<!-- start insert-links -->.*<!-- end insert-links -->"
-                appendContentOnMatchOnly: true
-                regexFlags: is
-                content: "<!-- start insert-links -->\n${{ steps.build_page_links.outputs.staging_links }}\n<!-- end insert-links -->"
-                token: ${{ secrets.GITHUB_TOKEN }}
+  get-pr-changes:
+    name: Get Changed Files & Update PR Description
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      contents: write
+      pull-requests: write
+      repository-projects: write
+    steps:
+      - uses: actions/checkout@v4
+      - name: Get Changed Files
+        id: changed-files
+        # pin to a specific commit to ensure stability
+        uses: tj-actions/changed-files@c65cd883420fd2eb864698a825fc4162dd94482c
```

Comment:
Specifying the version (v44) ensures stability, too, and I don't usually see people referencing a specific commit id. I'd revert this.

Reply:
This was new to me too. I found it in the [GitHub docs on hardening Actions.](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions). I think it's an extra layer of security against 1) unknown/unverified maintainers, 2) bad actors who gain access, and 3) obfuscated malicious code that slips into a version.

That said, I understand that it's less convenient. I could go either way but wanted to share why I changed it.

Reply:
Well that makes sense. According to All Things Git Holy, a tag is pinned to a specific commit and is immutable, but through some wrangling, it is possible for the repo owner to change it (https://stackoverflow.com/questions/21459540/add-new-commit-to-the-existing-git-tag). I suppose another approach would be to fork the repo and make it our own...but that's probably overkill. 
In short, keep it or revert it, and I'm happy. :)